File: youtube_summarizer/summarizer_openrouter.py
"""OpenRouter transcript summarization for deployment runtime."""

from typing import cast

from langchain_core.messages import HumanMessage, SystemMessage

from .llm_harness import ChatOpenRouter
from .prompts import get_langchain_summary_prompt
from .schemas import Summary
from .settings import get_settings


def summarize_video(
    transcript: str,
    target_language: str | None = None,
) -> Summary:
    settings = get_settings()
    clean_transcript = transcript.strip()
    if not clean_transcript:
        raise ValueError("Transcript cannot be empty")

    llm = ChatOpenRouter(
        model=settings.openrouter_summary_model,
        temperature=0,
        reasoning_effort=settings.openrouter_reasoning_effort,
        timeout=settings.llm_timeout_seconds,
    ).with_structured_output(Summary)

    messages = [
        SystemMessage(content=get_langchain_summary_prompt(target_language=target_language)),
        HumanMessage(content=f"Transcript:\n{clean_transcript}"),
    ]
    return cast(Summary, llm.invoke(messages))


async def summarize_video_async(
    transcript: str,
    target_language: str | None = None,
) -> Summary:
    settings = get_settings()
    clean_transcript = transcript.strip()
    if not clean_transcript:
        raise ValueError("Transcript cannot be empty")

    llm = ChatOpenRouter(
        model=settings.openrouter_summary_model,
        temperature=0,
        reasoning_effort=settings.openrouter_reasoning_effort,
        timeout=settings.llm_timeout_seconds,
    ).with_structured_output(Summary)

    messages = [
        SystemMessage(content=get_langchain_summary_prompt(target_language=target_language)),
        HumanMessage(content=f"Transcript:\n{clean_transcript}"),
    ]
    return cast(Summary, await llm.ainvoke(messages))

File: youtube-summarizer/summarizerOpenRouter.ts
import { HumanMessage, SystemMessage } from "@langchain/core/messages";
import { ChatOpenRouter } from "./llmClients.js";
import { getLangchainSummaryPrompt } from "./prompts.js";
import { type Summary, SummarySchema } from "./schemas.js";
import { getSettings } from "./settings.js";

export async function summarizerOpenRouter(
	transcript: string,
	targetLanguage: string | null = null,
): Promise<Summary> {
	const settings = getSettings();
	const cleanTranscript = transcript.trim();

	if (!cleanTranscript) {
		throw new Error("Transcript cannot be empty");
	}

	const model = settings.openrouterSummaryModel;
	const llm = ChatOpenRouter({
		model,
		reasoningEffort: settings.openrouterReasoningEffort,
		temperature: 0,
		timeoutMs: settings.llmTimeoutSeconds * 1000,
	}).withStructuredOutput(SummarySchema);

	const messages = [
		new SystemMessage(getLangchainSummaryPrompt(targetLanguage)),
		new HumanMessage(`Transcript:\n${cleanTranscript}`),
	];

	return await llm.invoke(messages);
}
